src/nomad_simulations/schema_packages/numerical_settings.py

```diff
@@ -890,15 +911,295 @@ def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        """,
+    )
+
+    differential_orders = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Derivative order of the radial wavefunction term.
+        """,
+    )  # TODO: add check non-negative # ? to remove
+
+    boundary_orders = Quantity(
+        type=np.int32,
+        shape=['n_terms'],
+        description="""
+        Differential order to which the radial wavefunction is matched at the boundary.
+        """,
+    )
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        if np.any(np.isneginf(self.differential_orders)):
+            logger.error('`APWLOrbital.differential_order` must be non-negative.')
+        if np.any(np.isneginf(self.boundary_orders)):
+            logger.error('`APWLOrbital.boundary_orders` must be non-negative.')
+
+
+class APWLChannel(ArchiveSection):
+    """
+    Channel of the angular momentum ($l$),
+    containing all orbitals that contribute to its description.
+    """
+
+    name = Quantity(
+        type=np.int32,
+        description="""
+        Angular momentum quantum number of the local orbital.
+        """,
+    )
+
+    n_wavefunctions = Quantity(
+        type=np.int32,
+        description="""
+        Number of wavefunctions in the l-channel, i.e. $(2l + 1) n_orbitals$.
+        """,
+    )
+
+    orbitals = SubSection(sub_section=APWWavefunction.m_def, repeats=True)
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        # self.name = self.m_def.name
+        self.n_wavefunctions = len(self.orbitals) * (2 * self.name + 1)
+
+
+class MuffinTinRegion(LocalizedBasisSet, Mesh):
+    """
+    Muffin-tin region around atoms, containing the augmented part of the APW basis set.
+    """
+
+    # there are 2 main ways of structuring the APW basis set
+    # either as APW and lo in the MT region
+    # or by l-channel in the MT region
+
+    l_max = Quantity(
+        type=np.int32,
+        description="""
+        Maximum angular momentum quantum number that is sampled.
+        Starts at 0.
+        """,
+    )
+
+    l_channels = SubSection(sub_section=APWLChannel.m_def, repeats=True)
+
+    def normalize(self, archive: 'EntryArchive', logger: 'BoundLogger') -> None:
+        super().normalize(archive, logger)
+        self.type = 'Spherical'
+
+
+class BasisSetContainer(NumericalSettings):
+    """Section providing an overview of the full basis set.
+    The basis set may contain multiple basis sets specifications `BasisSet`,
+    each with their own parameters."""
+
+    native_tier = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        Code-specific tag indicating the precision used
+        for the basis set and meshes of numerical routines.
+        """,
+    )
+
+    electronic_structure = Quantity(
```

Comment:
You can delete this. The `NumericalSettings` is a sub-section of the `ModelMethod`, so it will have this information above anyways.

Reply:
Not sure if I understand. Can you give me some examples of this string?

Reply:
They used to be `wavefunction` or `electron_density`. They may also cover [_augmentation charges_](https://www.vasp.at/wiki/index.php/ADDGRID), exact [exchange integrals](https://www.vasp.at/wiki/index.php/PRECFOCK), etc.
Essentially, a code can freely generate grids and basis sets for any specific case, e.g. https://wiki.fysik.dtu.dk/gpaw/documentation/core.html#uniform-grids

Reply:
Ok, I might understand that you wanna differentiate something like `scf_observable` or something like that for WFs or densities, but augmentation charges could go into its own quantity, and the same for how to treat the Hartree part, no?

My main problem is on the name of the quantity, "electronic_structure"

Reply:
I've already commented it out (will push soon), so we can return to it later on.

I'm fine with a different name, just not sure what to call it. The main examples are representations of the electronic structure, but the more exotic cases are indeed vaguer.

The reason I try to place them all here, is that they can be represented by the same (basis set) approach.
In the VASP cases that I linked, they are just grids. The HF part (_overlap densities_ actually) is still handled as plane waves, just with a coarser spacing. It even has it own tier system! Some codes do something similar for the XC density too. Augmentation charges are a necessary addition to the basis set when dealing with PAW pseudopotentials.

Don't get me wrong, though, each addition of a representation should be considered by itself.